GliaWidgets/Public/Glia/Glia+StartEngagement.swift

```diff
@@ -22,17 +23,29 @@ extension Glia {
     ///
     public func startEngagement(
         engagementKind: EngagementKind,
+        in queueIds: [String],
         theme: Theme = Theme(),
         features: Features = .all,
         sceneProvider: SceneProvider? = nil
     ) throws {
+        // `interactor?.queueIds.isEmpty == false` statement is needed for integrators who uses old interface
+        // and pass queue identifier throught `configuration` function.
+        guard !queueIds.isEmpty || interactor?.queueIds.isEmpty == false else { throw GliaError.startEngagementWithNoQueueIds }
         guard engagement == .none else { throw GliaError.engagementExists }
         guard let interactor = self.interactor else { throw GliaError.sdkIsNotConfigured }
         if let engagement = environment.coreSdk.getCurrentEngagement(),
             engagement.source == .callVisualizer {
             throw GliaError.callVisualizerEngagementExists
         }
 
+        // This check is needed for integrators who uses old interface
+        // and pass queue identifier throught `configuration` function,
+        // but would not pass queue ids in this method, so SDK would not override
+        // existed queue id.
+        if !queueIds.isEmpty {
```

Comment:
To be honest I don't really like that when it's time to remove the deprecated methods, we need to probably change this code and the guard. Wouldn't it be easier to just have like, the whole deprecated method works on its own without assumptions, and this one works as if the old one doesn't exist at all? Otherwise it will be like "this method works in X way, unless you are using deprecated methods, in which case it works in Y way". And the behavior of startEngagement vs configure + start is already very confusing for integrators. This has the pontential of making it worse.

Reply:
Agree with Gerson. It must be overcomplicated for integrator to cope with these logic.

Reply:
@gersonnoboa @EgorovEI 
The main reason of this code is that we changed type for `queueId` inside of Interactor and requires to have queue id(s) on start engagement step. Old configure method expects queueId in signature, however, a new interface doesn't expect it in configure, however, it requires for `startEngagement`.
To keep backward compatibility we need to handle
- queue id passed through `configure` method - old approach
- queue id not passed through `configure` method - new approach
- queue id not passed through `startEngagement` but in this case SDK should not trying start engagement without queue id at all
- queue id passed through `startEngagement` and in this case SDK should initiate a new engagement with these queue id(s) only

So, this not so clear checks needed because of backward compatibility and avoid possible mistakes during integrations.
Would it make sense or would you like to have a call and have a discussion to keep backward compatibility and solve this issue?

Reply:
If it will simplify the process of upgrading SDK and will not confuse integrators, than we can leave it as is. But I would have a call to understand this part better to minimize possible breaks that I (or another) can bring to this code.